### PR TITLE
Cleaned Up taskt.Core.Automation.ApplicationSettings.cs File

### PR DESCRIPTION
**Pull request by openbots-ff**
Cleaned up taskt.Core.Automation.ApplicationSettings and split classes into their own files.

### CODE REVIEW

**Review by CameronHerwig — DISMISSED** [review]

**Review by ob-ab — DISMISSED** [review]
Looks fine

File: taskt/Core/Settings/EngineSettings.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    public class EngineSettings
+    {
+        public bool ShowDebugWindow { get; set; }
+        public bool AutoCloseDebugWindow { get; set; }
+        public bool ShowAdvancedDebugOutput { get; set; }
+        public bool CreateMissingVariablesDuringExecution { get; set; }
+        public bool TrackExecutionMetrics { get; set; }
+        public string VariableStartMarker { get; set; }
+        public string VariableEndMarker { get; set; }
+        public System.Windows.Forms.Keys CancellationKey { get; set; }
```
Comment (melugi) [inline taskt/Core/Settings/EngineSettings.cs]: Can you shorten this reference?

File: taskt/Core/Settings/EngineSettings.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        public EngineSettings()
+        {
+            ShowDebugWindow = true;
+            AutoCloseDebugWindow = true;
+            ShowAdvancedDebugOutput = false;
+            CreateMissingVariablesDuringExecution = true;
+            TrackExecutionMetrics = true;
+            VariableStartMarker = "{";
+            VariableEndMarker = "}";
+            CancellationKey = System.Windows.Forms.Keys.Pause;
```
Comment (melugi) [inline taskt/Core/Settings/EngineSettings.cs]: Shorten this reference as well.

**Review by melugi — CHANGES_REQUESTED** [review]
Looks good, there were two small changes that need to be made. Thanks!

File: taskt/Core/Settings/EngineSettings.cs (hunk 2 above)
Comment (openbots-ff) [reply taskt/Core/Settings/EngineSettings.cs]: done

File: taskt/Core/Settings/EngineSettings.cs (hunk 1 above)
Comment (openbots-ff) [reply taskt/Core/Settings/EngineSettings.cs]: done

**Review by melugi — DISMISSED** [review]

**Review by melugi — DISMISSED** [review]

File: taskt/Core/Settings/WhiteListIPSettings.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+
+namespace taskt.Core.Settings
+{
+    [Serializable]
+    public class WhiteListIPSettings
+    {
+        public string Value { get; set; }
+        public WhiteListIPSettings(string s)
```
Comment (ob-ab) [inline taskt/Core/Settings/WhiteListIPSettings.cs]: Can you please change this parameter name `string s` to a meaningful variable name.

**Review by ob-ab — COMMENTED** [review]
Please make a small change of renaming Parameter name to a meaningful variable name.
Everything else looks good to me.

File: taskt/Core/Settings/ServerSettings.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        public bool RetryServerConnectionOnFail { get; set; }
+        public bool BypassCertificateValidation { get; set; }
+        public string ServerURL { get; set; }
+        public string ServerPublicKey { get; set; }
+        public string HTTPServerURL { get; set; }
+        public Guid HTTPGuid { get; set; }
+
+        public ServerSettings()
+        {
+            HTTPServerURL = "https://localhost:44377/";
```
Comment (CameronHerwig) [inline taskt/Core/Settings/ServerSettings.cs:22]: Long term we should move this into an app config for default values. Short term this is fine.

**Review by CameronHerwig — APPROVED** [review]

**Review by ob-ab — APPROVED** [review]
Looks fine.

**Pull request merged**